### PR TITLE
Refactor Home page and replace current index /quests

### LINKED ISSUE #951: Refactor Home page and replace current index  `/quests`
## Description 📹

Create a new index page following [this figma](https://www.figma.com/design/fh0OAvj4AS08kHoSxu3DkE/%F0%9F%9A%80-Starknet-Quest?node-id=6294-2187&m=dev)

Capture d’écran 2024-11-17 à 14 26 02

Rename the current index to `/quests` and adapt links of the navbar

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:
   ```bash
   git checkout -b fix-[issue-number]
   ```

2. **Implement Changes**:  
File Structure Changes:
- [ ] Create new directory `app/quests`
- [ ] Move current `app/page.tsx` to `app/quests/page.tsx`
- [ ] Create new `app/page.tsx` for the landing page

Navigation Updates:
- [ ] Update navbar links to point to new `/quests` route

New Landing Page Implementation:
- [ ] Create new components for landing page sections:
  - [ ] Hero section
  - [ ] Features grid
  - [ ] Call-to-action section
- [ ] Implement responsive design following Figma specs
- [ ] Add animations and transitions as shown in design
- [ ] Ensure proper image optimization for new assets

Testing:
- [ ] Test all navigation flows
- [ ] Verify responsive behavior
- [ ] Check accessibility
- [ ] Ensure no broken links

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:
   ```bash
   git commit -m "Fix: [Short description of the fix]"
   ```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing `Close #[issue_id]`.
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by baitcode**
## Task list
- [x] Create new directory app/quests
- [x] Move current app/page.tsx to app/quests/page.tsx
- [x] Create new app/page.tsx for the landing page
- [x] Update navbar links to point to new /quests route
- [x] Create new components for landing page sections:
    - [x] Hero section
    :x: Features grid
    - [x] Call-to-action section
- [x] Implement responsive design following Figma specs
- [x] Add animations and transitions as shown in design
- [x] Ensure proper image optimization for new assets
- [x] remove TODO(baitcode)

Testing 

- [x] Test all navigation flows
- [x] Verify responsive behavior
- [x] Check accessibility
- [x] Ensure no broken links

Fixed [dependency resolution problem](https://github.com/lfglabs-dev/starknet.quest/issues/951#issuecomment-2495894314)

Moved old page, created new one.
Added animation library Motion.
Implemented requested effect. To implement effect I had to cut svg's, as I don't have an expensive software I came up with the idea to use svg's clip-path functionality, and [generated](https://gist.github.com/baitcode/12c74812b45f28ba1b699c0c55dc0142) clip-paths for every svg.

- Feature
Resolves: #951 

## Other information
Script for split generation: [GIST](https://gist.github.com/baitcode/12c74812b45f28ba1b699c0c55dc0142)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new user interface for displaying quests, including a featured quest banner and navigation options.
	- Added components for quests and categories, statistics display, and user participation guidance.

## Release Notes

- **New Features**
  - Introduced a new `Page` component for displaying quests, featuring a banner, tabs, and statistics.
  - Added `CallToAction`, `Card`, `Hero`, and `SplitTitle` components for enhanced user interaction and visual appeal.

- **UI Enhancements**
  - Updated the `Navbar` with a new "DeFi" link and improved notification handling.
  - Enhanced styling and responsiveness across various components, including cards and navbar.

- **Bug Fixes**
  - Updated import paths for wallet-related functions to ensure compatibility with new modules.

- **Chores**
  - Updated package dependencies and Tailwind CSS configuration for improved performance and styling capabilities.

### CODE REVIEW

File: app/page.tsx
Comment (fricoben) [inline app/page.tsx]: delete {``} to use ""

File: components/UI/navbar.tsx
Comment (fricoben) [reply components/UI/navbar.tsx]: Yep agree !

File: components/UI/navbar.tsx
Comment (fricoben) [reply components/UI/navbar.tsx:208]: Agree ! Please propose changes to our contrib

File: components/pages/home/splitTitle.tsx
Comment (fricoben) [inline components/pages/home/splitTitle.tsx:21]: To add in another file as an SVG to `/icons`

**Review by fricoben — CHANGES_REQUESTED** [review]
Have a look at AI suggestion, Code doesn't compile, please make sure it compiles before sending the PR

File: components/pages/home/splitTitle.tsx
Comment (baitcode) [reply components/pages/home/splitTitle.tsx:21]: Pardon. I do not understand what are you trying to say.

File: components/pages/home/splitTitle.tsx
Comment (baitcode) [reply components/pages/home/splitTitle.tsx:392]: But that's the whole point! 8 )

File: components/pages/home/splitTitle.tsx
Comment (baitcode) [reply components/pages/home/splitTitle.tsx:3]: nope

File: components/UI/navbar.tsx
Comment (baitcode) [reply components/UI/navbar.tsx:208]: Can you create a separete issue for that? Changes are too huge and out of scope of that task.

**Comment by baitcode** [comment]
>> Have a look at AI suggestion, Code doesn't compile, please make sure it compiles before sending the PR

I've checked the issue description and contribution guidelines. I don't understand what do you mean by `compilation error`. `npm build` - does not produce error. `npm dev` - runs devserver without any problem. CI workflows are blocked by your approval, so I don't see what is going on there. If you have Idea how am I supposed to check for that, please let me know. Thank you!

Ok. I guess it has something to do with the linter.

File: eslint.config.mjs
Comment (baitcode) [reply eslint.config.mjs]: nope. leaving htis to maintainer

File: app/page.tsx
Comment (baitcode) [reply app/page.tsx:34]: nope. It's out of scope.

**Review by fricoben — COMMENTED** [review]
Capture d’écran 2024-11-28 à 09 45 32

Title + description are not super fit with responsive. Fix this and we're good to merge !

**Review by fricoben — APPROVED** [review]
lgtm

**Comment by baitcode** [comment]
@all-contributors please add baitcode for new home page design

**Comment by baitcode** [comment]
Great job

**Pull request merged**
